src/crypto/crypto_mbedtls_alt.c

```diff
@@ -605,10 +605,57 @@ int sha384_prf(
     return hmac_prf_bits(key, key_len, label, data, data_len, buf, buf_len * 8, MBEDTLS_MD_SHA384);
 }
 
+/**
+ * Based on Supplicant internal implementaion of SHA-256. This API
+ * uses PSA APIs instead of Supplicant internal implementation or
+ * mbedtls APIs.
+ */
+static int hmac_prf256(const u8 *key,
+                       size_t key_len,
+                       const char *label,
+                       const u8 *data,
+                       size_t data_len,
+                       u8 *buf_in,
+                       size_t buf_len_bits,
+                       mbedtls_md_type_t md_type)
+{
+    unsigned short ctr, n_le = host_to_le16(buf_len_bits);
+    const u8 *addr[]         = {(u8 *)&ctr, (u8 *)label, data, (u8 *)&n_le};
+    const size_t len[]       = {2, os_strlen(label), data_len, 2};
+    size_t buf_len           = (buf_len_bits + 7) / 8;
+    u8 *buf  = buf_in;
+
+    for (ctr = 1; buf_len >= SHA256_MAC_LEN; buf_len -= SHA256_MAC_LEN, ++ctr)
+    {
+            if (hmac_sha256_vector(key, key_len, 4, addr, len, buf))
```

Comment:
this seems not using addr[i] as input.

Reply:
Kindly refer `sha1_t_prf` or `sha1_prf` in the same file. Also, the internal implementation of sha256 PRF. when the hmac API is called, the number of elements is passed as a parameter to the relevant API. In this case which is 4 elements in the array addr. I have followed the same. The output for the WPA2-PSK-256 connection passed and the test vectors tested also provided desired result